### ISSUE TITLE
Trivy is not detecting OS

### ISSUE THREAD

**Issue by Yogesh-BK** [aquasecurity/trivy#3006]
## Description
We had installed trivy in our aws agents where Debian OS in installed and when we try to run trivy scan, we are getting Detected OS: unknown and trivy is not performing the scan
The command that we are using to run trivy scan is  ``` trivy fs <folder name>``` and we are getting the same error when we try to run ```trivy repo <repo-url>```

## What did you expect to happen?
It should detect the OS and perform both secret and vulnerability scans

## What happened instead?
It is giving the below errors
        INFO	Detected OS: unknown
        INFO	Number of PL dependency files: 0

## Output of run with `-debug`:

```
DEBUG	Severities: UNKNOWN,LOW,MEDIUM,HIGH,CRITICAL
DEBUG	cache dir:  /root/.cache/trivy
DEBUG	DB update was skipped because DB is the latest
DEBUG	DB Schema: 1, Type: 1, UpdatedAt: 2022-10-11 00:52:50.469860837 +0000 UTC, NextUpdate: 2022-10-11 06:52:50.469860337 +0000 UTC, DownloadedAt: 2022-10-11 04:40:55.884378489 +0000 UTC
DEBUG	Vulnerability type:  [os library]
DEBUG	Analysis error: unable to parse update_code.sh: failed to parse update_code.sh: unrecognized executable format
DEBUG	OS is not detected and vulnerabilities in OS packages are not detected.
INFO	Detected OS: unknown
INFO	Number of PL dependency files: 0
```

## Output of `trivy -v`:

```
Version: 0.18.3
Vulnerability DB:
  Type: Light
  Version: 1
  UpdatedAt: 2022-10-11 00:52:50.469860837 +0000 UTC
  NextUpdate: 2022-10-11 06:52:50.469860337 +0000 UTC
  DownloadedAt: 2022-10-11 04:40:55.884378489 +0000 UTC
```

## Additional details (base image name, container registry info...):
We are trying to scan a public repository and not the images (And the repository is quite a large repository)

**Comment by DmitriyLewen** [comment]
Hello @Yogesh-BK 
Thanks for your report!

To detect OS Trivy needs os information file(e.g. `/etc/os-release`).
In most cases repositories don't have this file.
If you want to find OS for `fs` command, you need to scan root directory.
e.g.
```bash
root@e28a57860197:/# trivy rootfs /
2022-10-12T03:52:10.042Z	INFO	Vulnerability scanning is enabled
2022-10-12T03:52:10.043Z	INFO	Secret scanning is enabled
2022-10-12T03:52:10.043Z	INFO	If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2022-10-12T03:52:10.043Z	INFO	Please see also https://aquasecurity.github.io/trivy/v0.32/docs/secret/scanning/#recommendation for faster secret detection
2022-10-12T03:52:13.014Z	INFO	Detected OS: ubuntu
2022-10-12T03:52:13.014Z	INFO	Detecting Ubuntu vulnerabilities...
2022-10-12T03:52:13.015Z	INFO	Number of language-specific files: 0

e28a57860197 (ubuntu 20.04)

Total: 85 (UNKNOWN: 0, LOW: 49, MEDIUM: 36, HIGH: 0, CRITICAL: 0)
```

You also need to understand that there are to commands: `fs` and `rootfs`.
These commands scan different language packages.
More information [here](https://aquasecurity.github.io/trivy/v0.32/docs/vulnerability/detection/language/).

##

About secrets scan:
If i remember correctly: `v0.18` can't find secrets.
You need [to update](https://aquasecurity.github.io/trivy/v0.32/getting-started/installation/) your Trivy to scan secrets. 

Regards, Dmitriy

**Comment by lecrepont01** [comment]
Hi, I am having a similar issue while trying to scan a specific dir like `trivy fs ./mydir`.
Actually I am just running a local test on my macOS before putting some secret detection in my CI.

Version of trivy : 0.41.0

My output is slightly different though :

```
2023-05-02T11:02:50.710+0200    INFO    Loaded trivy.yaml
2023-05-02T11:02:50.712+0200    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-05-02T11:02:50.723+0200    DEBUG   cache dir:  /Users/.../Library/Caches/trivy
2023-05-02T11:02:50.723+0200    INFO    Secret scanning is enabled
2023-05-02T11:02:50.723+0200    INFO    If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-05-02T11:02:50.723+0200    INFO    Please see also https://aquasecurity.github.io/trivy/v0.41/docs/secret/scanning/#recommendation for faster secret detection
2023-05-02T11:02:50.723+0200    INFO    Loading trivy-secret.yaml for secret scanning...
2023-05-02T11:02:50.724+0200    DEBUG   Walk the file tree rooted at 'zfixtures' in parallel
2023-05-02T11:02:50.724+0200    DEBUG   Skipped secret scanning on "private.pem" as non-compliant to the rule "dummy_rule"
2023-05-02T11:02:50.744+0200    DEBUG   OS is not detected.
```

No detection is happening although I injected some fake secrets in the dir that should trigger trivy secrets.

**Comment by DmitriyLewen** [comment]
Hello @lecrepont01 
Thanks for your report!

You expected Trivy to find the secret, right?
It doesn't seem to be the same issue.

Can you create a new issue with information about your case.
It will be great if you send an example to reproduce your error.

Can you also send your `trivy-secret.yaml` file? It looks like you are skipping secrets because of `dummy_rule`.

Regards, Dmitiry

**Comment by lecrepont01** [comment]
Thanks for your help @DmitriyLewen. I opened issue #4157.
Best regards,
Leo
